File: semgrep-rules/go/cors/testdata/cors_origin_reflection.go
package testdata

import "net/http"

// ruleid: go-cors-origin-reflection
func reflectOriginInline(w http.ResponseWriter, r *http.Request) {
	w.Header().Set("Access-Control-Allow-Origin", r.Header.Get("Origin"))
}

// ruleid: go-cors-origin-reflection
func reflectOriginVariable(w http.ResponseWriter, r *http.Request) {
	origin := r.Header.Get("Origin")
	w.Header().Set("Access-Control-Allow-Origin", origin)
}

// ok: go-cors-origin-reflection
func staticOrigin(w http.ResponseWriter, r *http.Request) {
	w.Header().Set("Access-Control-Allow-Origin", "https://trusted.example.com")
}

// ok: go-cors-origin-reflection
func wildcardOrigin(w http.ResponseWriter, r *http.Request) {
	w.Header().Set("Access-Control-Allow-Origin", "*")
}

File: semgrep-rules/java/cors/testdata/CorsOriginReflection.java
package com.example.testdata;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

public class CorsOriginReflection {

    // ruleid: java-cors-origin-reflection
    public void reflectOriginInline(HttpServletRequest request, HttpServletResponse response) {
        response.setHeader("Access-Control-Allow-Origin", request.getHeader("Origin"));
    }

    // ruleid: java-cors-origin-reflection
    public void reflectOriginVariable(HttpServletRequest request, HttpServletResponse response) {
        String origin = request.getHeader("Origin");
        response.setHeader("Access-Control-Allow-Origin", origin);
    }

    // ok: java-cors-origin-reflection
    public void staticOrigin(HttpServletRequest request, HttpServletResponse response) {
        response.setHeader("Access-Control-Allow-Origin", "https://trusted.example.com");
    }

    // ok: java-cors-origin-reflection
    public void wildcardOrigin(HttpServletRequest request, HttpServletResponse response) {
        response.setHeader("Access-Control-Allow-Origin", "*");
    }
}
